packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -235,7 +235,17 @@ class ChromiumLauncher {
     ), skipCheck);
   }
 
-  Future<Process?> _spawnChromiumProcess(List<String> args) async {
+  Future<Process?> _spawnChromiumProcess(List<String> args, String chromeExecutable) async {
+    if (_operatingSystemUtils.hostPlatform == HostPlatform.darwin_arm) {
+      final ProcessResult result = _processManager.runSync(<String>['file', chromeExecutable]);
+      // Check if ARM Chrome is installed.
```

Comment:
When you install Chrome on a Mac it asks which architecture you want to install, and it will happily run in Rosetta if you choose "Mac with Intel Chip".  So don't assume ARM macs have ARM Chrome installed, and check.

<img width="593" alt="Screen Shot 2021-04-19 at 4 45 52 PM" src="https://user-images.githubusercontent.com/682784/115316703-b6a89080-a12e-11eb-8e6f-6655700a75c6.png">